src/account.ts

```diff
@@ -404,12 +407,17 @@ export class Account {
             },
         });
         let result;
-        result = await this.client.tvm.run_executor({
-            account: accountForExecutorAccount(await this.boc()),
-            abi: this.abi,
-            message: message.message,
-        });
-        return result.fees;
+        const boc = await this.boc();
+        if (typeof boc === "string" && boc !=="") {
```

Comment:
I think it is better to move this comparison and error throwing to the `boc()` function. And remove `| null` from `boc()` function return type.

Reply:
It could be great, 
The key question here is: when the boc() function sees an empty BOC, is that an exception or a valid result?
I'm assuming that BOC=null is a valid result telling us that the account "NonExist" (are there other situations where BOC can be empty? I don't know....)

if we consider  that empty BOC is an exception, we need to assign some code to distinguish this error from others (for example  network errors)

Reply:
When user wants BOC, he wants required non empty BOC. When user's logic depends on BOC presence, then the user have to check an account type before he gets the boc.

Reply:
So.... what is the solution than ?
If you want boc() function to throw  an error in the case of an empty BOC (account is found but has type "notExists")  we need to assign some new error code.